m4opt/models/extinction/airmass.py

```diff
@@ -0,0 +1,394 @@
+from ..core import state
+from .core import BaseExtinction
+
+import numpy as np
+import astropy.units as u
+from astropy.coordinates import EarthLocation, AltAz
+from astropy.modeling import custom_model
+from astropy.modeling.models import Const1D, Tabular1D
+from astropy.table import QTable
+from astropy.utils.data import download_file
+
+
+path_to_extin = ('https://raw.githubusercontent.com/astropy/'
+                 'specreduce-data/main/specreduce_data/'
+                 'reference_data/extinction/')
+
+__all__ = ('Airmass', 'AtmosphericExtinction',
+           'PlaneParallelAirmass', 'KastenYoungAirmass')
+
+
+def read_extinction_table(placename):
+    """
+    extinction tables usually have
+    x in angstroms
+    y in mag per airmass
+    """
+    try:
+        tablename = AtmosphericExtinction.available_tables[placename]
+    except KeyError:
+        raise ValueError("Invalid placename {0}. Must be one of {1}".format(
+                        placename,
+                        AtmosphericExtinction.available_tables.keys()))
```

Comment:
Suggest using an f-string.

Reply:
Whoa, just learned about f-strings! Yeah, that's definitely easier.